planner/core/planbuilder.go

```diff
@@ -1141,6 +1145,20 @@ func removeIgnoredPaths(paths, ignoredPaths []*util.AccessPath, tblInfo *model.T
 	return remainedPaths
 }
 
+func removeTiflashDuringStaleRead(paths, ignoredPaths []*util.AccessPath) []*util.AccessPath {
+	if len(ignoredPaths) == 0 {
+		return paths
+	}
+	remainedPaths := make([]*util.AccessPath, 0, len(paths))
```

Comment:
It is possible to do in-place filter in golang. See https://github.com/golang/go/wiki/SliceTricks.

Reply:
updated.